packages/features/ee/impersonation/lib/ImpersonationProvider.ts

```diff
@@ -113,6 +117,69 @@ export function checkGlobalPermission(session: Session | null) {
   }
 }
 
+/**
+ * Check PBAC permissions for impersonation
+ * This function integrates with the new PBAC system to determine impersonation permissions
+ */
+export async function checkPBACImpersonationPermission({
+  userId,
+  teamId,
+  userRole,
+  organizationId,
+}: {
+  userId: number;
+  teamId?: number;
+  userRole: MembershipRole;
+  organizationId?: number | null;
+}): Promise<boolean> {
+  try {
+    // For organization-level impersonation
+    if (organizationId) {
+      const orgPermissions = await getSpecificPermissions({
+        userId,
+        teamId: organizationId,
+        resource: Resource.Organization,
+        userRole,
+        actions: [CustomAction.Impersonate],
+        fallbackRoles: {
+          [CustomAction.Impersonate]: {
+            roles: [MembershipRole.ADMIN, MembershipRole.OWNER],
+          },
+        },
+      });
+
+      if (orgPermissions[CustomAction.Impersonate]) {
+        return true;
+      }
+    }
+
+    // For team-level impersonation
+    if (teamId) {
```

Comment:
I wonder what happens if... organizationId exists, but `orgPermissions[CustomAction.Impersonate]` is false, so it didn't return, but somehow teamId also exists, and it goes into this if branch. Is it intended? Or do we want something like this?

```suggestion
    } else if (teamId) {
      // For team-level impersonation
```